### PR TITLE
calc_error.py housekeeping

### PR DESCRIPTION
**Pull request by aqitya**
Fixing exception and warnings messages as well as reformatting function parameters and documentation.

### CODE REVIEW

File: src/prog_models/utils/calc_error.py [hunk 1]
```diff
@@ -125,18 +127,19 @@ def RMSE(m, times, inputs, outputs, **kwargs):
     return np.sqrt(MSE(m, times, inputs, outputs, **kwargs))
 
 
-def MSE(self, times, inputs, outputs, **kwargs) -> float:
+def MSE(m, times, inputs, outputs, **kwargs) -> float:
```
Comment (mstraut) [inline src/prog_models/utils/calc_error.py:130]: What was the reason to make this change?

**Review by mstraut — CHANGES_REQUESTED** [review]
The changes look good and make sense, I have a couple of things I'd like to request...

change the import numpy as np to import from numpy.. just what is being used. No need to import all of numpy.

Adding type hints will help clarify the code.
https://docs.python.org/3/library/typing.html

Housekeeping is so important, thanks for taking the time!

File: src/prog_models/utils/calc_error.py (hunk 1 above)
Comment (aqitya) [reply src/prog_models/utils/calc_error.py:130]: Other calc_error methods have defined the model to be 'm', so for the sake of consistency we changed 'self' to 'm'.

For a similar reason, I might have to include the '-> float' to all the calc_methods or maybe remove the '-> float'.

Comment (mstraut) [reply src/prog_models/utils/calc_error.py:130]: ok, was just trying to understand if there was a reason for it. sounds fine. 

"For a similar reason, I might have to include the '-> float' to all the calc_methods or maybe remove the '-> float'." keep those for clarification. One of my requests for housekeeping is type hints. It's something that's inconsistent in ProgPy and will help future developers when contributing.

**Review by kjjarvis — APPROVED** [review]
Nice housekeeping additions, and good change of making the default time step more reasonable

**Review by kjjarvis — APPROVED** [review]
Nice housekeeping additions, and good change of making the default time step more reasonable

**Comment by aqitya** [comment]
@mstraut Great suggestion regarding the hints for the functions.

Also, we end up using 8 or so function calls from numpy, so I believe it would probably be best to keep it as it is right now. However, I am not sure where to draw the line between having it explicitly defined in the import statement...

File: src/prog_models/utils/calc_error.py [hunk 2]
```diff
@@ -97,7 +99,7 @@ def MAX_E(m, times, inputs, outputs, **kwargs):
     return err_max
 
 
-def RMSE(m, times, inputs, outputs, **kwargs):
+def RMSE(m, times, inputs, outputs, **kwargs) -> float:
```
Comment (mstraut) [inline src/prog_models/utils/calc_error.py:102]:
```suggestion
def RMSE(m, times: list[float], inputs: list[dict], outputs: list[dict], **kwargs) -> float:
```

File: src/prog_models/utils/calc_error.py (hunk 1 above)
Comment (mstraut) [inline src/prog_models/utils/calc_error.py:130]:
```suggestion
def MSE(m, times: list[float], inputs: list[dict], outputs: list[dict], **kwargs) -> float:
```

File: src/prog_models/utils/calc_error.py [hunk 3]
```diff
@@ -274,16 +280,17 @@ def MAE(m, times, inputs, outputs, **kwargs):
...
-                    warn(f"Model unstable- NaN reached in simulation (t={t})")
+                    warn("Model unstable- NaN reached in simulation (t={})".format(t))
                     break
             err_total += np.sum(
                 np.abs(z.matrix - z_obs.matrix))
             counter += 1
     return err_total/counter
 
-def MAPE(m, times, inputs, outputs, **kwargs):
+def MAPE(m, times, inputs, outputs, **kwargs) -> float:
```
Comment (mstraut) [inline src/prog_models/utils/calc_error.py:293]:
```suggestion
def MAPE(m, times: list[float], inputs: list[dict], outputs: list[dict], **kwargs) -> float:
```

File: src/prog_models/utils/calc_error.py [hunk 4]
```diff
@@ -11,7 +11,7 @@
 import numpy as np
 
 
-def MAX_E(m, times, inputs, outputs, **kwargs):
+def MAX_E(m, times, inputs, outputs, **kwargs) -> float:
```
Comment (mstraut) [inline src/prog_models/utils/calc_error.py:14]:
```suggestion
def MAX_E(m, times: list[float], inputs: list[dict], outputs: list[dict], **kwargs) -> float:
```

File: src/prog_models/utils/calc_error.py [hunk 5]
```diff
@@ -146,64 +149,66 @@ def MSE(self, times, inputs, outputs, **kwargs) -> float:
...
+                    warn("Model unstable- NaN reached in simulation (t={})".format(t))
                     break
-            err_total += np.sum(np.square(z.matrix - z_obs.matrix), where=~np.isnan(z.matrix))
+            err_total += np.sum(np.square(z.matrix - z_obs.matrix), where= ~np.isnan(z.matrix))
             counter += 1
 
     return err_total/counter
 
-def MAE(m, times, inputs, outputs, **kwargs):
+def MAE(m, times, inputs, outputs, **kwargs) -> float:
```
Comment (mstraut) [inline src/prog_models/utils/calc_error.py:211]:
```suggestion
def MAE(m, times: list[float], inputs: list[dict], outputs: list[dict], **kwargs) -> float:
```

**Review by mstraut — CHANGES_REQUESTED** [review]
Hey, I made some suggestions, for type hints that we're trying to make sure are there for clarity. It takes away some of the confusion for new users and developers.

File: src/prog_models/prognostics_model.py [hunk 6]
```diff
@@ -172,21 +173,21 @@ def __setstate__(self, params: dict) -> None:
 
         states = self.states
 
-        class StateContainer(DictLikeMatrixWrapper):
+        class StateContainer(containers.StateContainer):
```
[the remark this answers is not in the document]
Comment (aqitya) [reply src/prog_models/prognostics_model.py:176]: Yeah, I think we did, I'll revert the changes back. However, I will still keep enough changes such that the TypeHints remain for calc_error and estimate_params.

File: src/prog_models/utils/calc_error.py [hunk 7]
```diff
@@ -6,12 +6,13 @@
...
 
 from collections.abc import Iterable
+from typing import List
 from warnings import warn
 import math
 import numpy as np
 
 
-def MAX_E(m, times, inputs, outputs, **kwargs):
+def MAX_E(m, times: List[float], inputs: List[dict], outputs: List[dict], **kwargs) -> float:
```
Comment (kjjarvis) [inline src/prog_models/utils/calc_error.py:15]:
```suggestion
def MAX_E(m, times: List[float], inputs: List[InputContainer], outputs: List[OutputContainer], **kwargs) -> float:
```
Is this consistent with our new notation? If so, let's add it here and for the other methods below

**Review by kjjarvis — APPROVED** [review]
Just one last suggestion - this looks great.

**Pull request merged**